client/src/pages/Room/Room.tsx

```diff
@@ -114,13 +118,21 @@ const Room: React.FC = () => {
       },
       updateQuery: (prev, { fetchMoreResult }) => {
         if (!fetchMoreResult) return prev;
-        return update(prev, {
+        const updatedData = update(prev, {
           messages: {
             messages: {
               $unshift: fetchMoreResult.messages?.messages,
             },
           },
         });
+
+        // scroll jumping fix
+        const lastMessage: any = bodyRef.current.getElementsByClassName(
+          "message__item"
+        )[0];
+        bodyRef.current.scrollTop = lastMessage.offsetTop - 70;
+
+        return updatedData;
```

Comment:
Steps to fix Scroll jumping on newMessage pagination data,

- Find the last Message DOM Element.
- set the Scroll to that lastMessage's offsetTop with a offset